celltype_annotation/analysis/SingleR-non-immune-ref-comparison.Rmd

```diff
@@ -0,0 +1,436 @@
+  tidyr::separate(reference, into = c("label_type", "reference"), sep = "_", extra = "merge") |>
+  # classify each label as pruned or not
+  dplyr::mutate(pruned = ifelse(is.na(pruned.labels), "Yes", "No"))
+```
+
+```{r fig.height=5}
+# plot the distribution of delta.next for each reference, faceting by label type (main, fine, ont)
+plot_metric_distribution(all_results, "delta.next")
+```
+
+```{r fig.height=5}
+# make the same plot but color by celltypes, looking only at main labels
+main_label_results <- all_results |>
+  dplyr::filter(label_type == "label.main") |> 
+  dplyr::mutate(top_pruned_labels = forcats::fct_lump_n(pruned.labels, 10))
+
+ggplot(main_label_results, aes(x = reference, y = delta.next, group = reference, color = top_pruned_labels)) +
+  ggforce::geom_sina(size = 0.1, alpha = 0.5) +
+  stat_summary(
+    aes(group = reference),
+    color = "black",
+    # median and quartiles for point range
+    fun = "median",
+    fun.min = function(x) {
+      quantile(x, 0.25)
+    },
+    fun.max = function(x) {
+      quantile(x, 0.75)
+    },
+    geom = "pointrange",
+    position = position_dodge(width = 0.9),
+    size = 0.1
+  ) +
+  guides(x = guide_axis(angle = 90),
+         colour = guide_legend(override.aes = list(size = 3, alpha = 1)))
+```
+In addition to the delta next, which only looks at the difference between the scores of the top two labels, below we will measure the difference between the top score and the median scores. 
+Then we will plot the distribution of this delta as well. 
+
+```{r}
+# function to calculate the median delta for each singleR result
+get_median_delta <- function(singler_result){
+  
+  # grab all the scores
+  scores <- singler_result$scores |>
+    as.matrix()
+  
+  # calculate the median delta (max score - median of all scores) 
+  median_delta <- rowMaxs(scores) - rowMedians(scores)
+  
+  # add median into the singler result and return as data frame
+  median_delta_result <- singler_result |>
+    as.data.frame() |>
+    dplyr::select(labels, delta.next, pruned.labels) |>
+    dplyr::mutate(median_delta = median_delta)
+  
+  return(median_delta_result)
+}
+```
+
+```{r}
+# get the median delta for each singler object and prep combined data frame for plotting distribution
+median_delta_results <- singler_results |> 
+  purrr::map(get_median_delta) |>
+  dplyr::bind_rows(.id = "reference") |>
+  tidyr::separate(reference, into = c("label_type", "reference"), sep = "_", extra = "merge") |>
+  # classify each label as pruned or not
+  dplyr::mutate(pruned = ifelse(is.na(pruned.labels), "Yes", "No"))
+```
+
+```{r, fig.height=5}
+# plot the distribution of delta.next for each reference, faceting by label type (main, fine, ont)
+plot_metric_distribution(median_delta_results, "median_delta")
+```
+
+The delta median appears to show more striking differences between the references containing cells we would expect in our dataset vs. references that are inappropriate. 
```

Comment:
Just chiming in here.. I guess a situation of `delta.next > median_delta` could result if the majority of the points _are_ the maximum value, and we could check if this is the case for distributions where some `delta.next` are higher than `median_delta`. And/or, it should be possible to do our own `delta.next` calculation if this is something we really want to understand. But, as Ally wrote in her comment below, we're likely headed to `median_delta` anyways, so an in-depth investigation here probably isn't super critical.

Reply:
I actually think that's a case where the `delta.next` would be lower still because the `delta.next` between the two high labels should be relatively close if they are both high, or at least I would think that. And with a median lower than the score for the second highest label then that value will still be higher. I think the only scenario in which `delta.next` is higher is if somehow the score for the 2nd highest label is below the score for the median which doesn't make sense to me at all.

Reply:
I can imagine something here if the scores used for delta calculation are not the 80% quartiles that are reported as the cell type scores, but 🤷‍♂️

Reply:
Oh, another thought is that the delta.next scores are based on the fine-tuning scores, which use a smaller set of genes, and may have higher max correlations. I bet that is it, come to think of it. Are the fine tuning scores reported at all?

Reply:
From what I can interpret, the `fine.tuning` scores are not reported. The docs on `rddr` say their should be a `tuned.scores` matrix, but I don't see that in the output and it's not in the docs on the github.